### PR TITLE
docs: add logging best practices

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

Updates code review checklist and contributing docs to include logging best practices.  

## Checklist
- [ ] Followed the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) when writing a release note.
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -46,6 +46,19 @@ a human. All code style enforcement must be automated to avoid bikeshedding
 and losing time.
 
 
+Logging
+=======
+
+The ddtrace logger should be used log events in the dd-trace-py library. Use `ddtrace.internal.logger.get_logger(__name__)` to initialize/retrieve an instance of the ddtrace logger (DDLogger).
+
+To ensure the ddtrace library produces consistent and secure logs the following best practices should be followed:
```
Comment (emmettbutler) [inline docs/contributing.rst:54]: I believe you need to add an extra linebreak before the bullet list to get it to format properly.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -41,4 +41,5 @@
 - [ ] Tests provided or description of manual testing performed is included in the code or PR.
 - [ ] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
 - [ ] All relevant GitHub issues are correctly linked.
-- [ ] Backports are identified and tagged with Mergifyio.
+- [ ] Change is easily debuggable and contains logs where appropriate.
+- [ ] Logs are meaningful, actionable and do not have the potentical to leak sensitive data.
```
Comment (Kyle-Verhoog) [inline .github/PULL_REQUEST_TEMPLATE.md:45]:
```suggestion
- [ ] Logs are meaningful, actionable and do not have the potential to leak sensitive data.
```

Comment (Kyle-Verhoog) [reply .github/PULL_REQUEST_TEMPLATE.md:45]: this applies to any kind of telemetry, not just logs

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -41,4 +41,5 @@
 - [ ] Tests provided or description of manual testing performed is included in the code or PR.
 - [ ] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
 - [ ] All relevant GitHub issues are correctly linked.
-- [ ] Backports are identified and tagged with Mergifyio.
+- [ ] Change is easily debuggable and contains logs where appropriate.
```
Comment (Kyle-Verhoog) [inline .github/PULL_REQUEST_TEMPLATE.md:44]: If we can avoid using "easily" to be more precise might be better. Can we hint as to how to make it "easy" to debug?

Also logs->telemetry?

File: docs/contributing.rst [hunk 4]
```diff
@@ -46,6 +46,19 @@ a human. All code style enforcement must be automated to avoid bikeshedding
...
+* Logs should be generated with the level DEBUG, INFO, WARNING, or ERROR.
+* Log messages are grammatically correct and do not contain spelling errors.
+* Log messages should be standalone and actionable. They should not require context from other logs, metrics or trace data.
+* Logging generic attributes like __dict__ is a serious potential source of data leaks. These kinds of generic log statements should therefore be avoided and replaced with more specific ones (e.g. logging the required attributes explicitly).
```
Comment (Kyle-Verhoog) [inline docs/contributing.rst:58]:
```suggestion
* Logging generic attributes like __dict__ is a serious potential source of data leaks. These kinds of generic log statements should therefore be avoided and replaced with more specific ones (e.g. logging the required attributes explicitly).
```

Maybe worth changing this point to say something like "Log data is sensitive and should not contain application secrets or other sensitive data"?

**Review by Kyle-Verhoog — COMMENTED** [review]
a few suggestions

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -41,4 +41,5 @@
 - [ ] Tests provided or description of manual testing performed is included in the code or PR.
 - [ ] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
 - [ ] All relevant GitHub issues are correctly linked.
-- [ ] Backports are identified and tagged with Mergifyio.
+- [ ] Change is easily debuggable and contains telemetry (logs, metrics, etc.) where appropriate.
```
Comment (Kyle-Verhoog) [inline .github/PULL_REQUEST_TEMPLATE.md:44]:
```suggestion
- [ ] Change contains telemetry (logs, metrics, etc.) where appropriate.
```

File: docs/contributing.rst [hunk 6]
```diff
@@ -46,6 +46,19 @@ a human. All code style enforcement must be automated to avoid bikeshedding
 and losing time.
 
 
+Logging
+=======
+
+The ddtrace logger should be used log events in the dd-trace-py library. Use `ddtrace.internal.logger.get_logger(__name__)` to initialize/retrieve an instance of the ddtrace logger (DDLogger).
```
Comment (Kyle-Verhoog) [inline docs/contributing.rst:52]:
```suggestion
The ddtrace logger should be used log events in the dd-trace-py library. Use ``ddtrace.internal.logger.get_logger(__name__)`` to initialize/retrieve an instance of the ddtrace logger (DDLogger).
```

**Review by Kyle-Verhoog — DISMISSED** [review]

File: docs/contributing.rst [hunk 7]
```diff
@@ -46,6 +46,19 @@ a human. All code style enforcement must be automated to avoid bikeshedding
 and losing time.
 
 
+Logging
+=======
+
+The ddtrace logger should be used log events in the dd-trace-py library. Use ``ddtrace.internal.logger.get_logger(__name__)`` to initialize/retrieve an instance of the ddtrace logger (DDLogger).
```
Comment (Yun-Kim) [inline docs/contributing.rst:52]:
```suggestion
The ddtrace logger should be used to log events in the dd-trace-py library. Use ``ddtrace.internal.logger.get_logger(__name__)`` to initialize/retrieve an instance of the ddtrace logger (DDLogger).
```

**Review by emmettbutler — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**